casepro/contacts/models.py

```diff
@@ -203,10 +215,11 @@ def suspend_groups(self):
     def restore_groups(self):
         with self.lock(self.org, self.uuid):
             for group in list(self.suspended_groups.all()):
-                self.groups.add(group)
-                self.suspended_groups.remove(group)
+                if not group.is_dynamic:
```

Comment:
We know contacts have been "suspended" from dynamic groups already which they shouldn't have